### PR TITLE
Level design - so god damn much changed

### LINKED ISSUE #64: Level Design
This will be for creating level boarders, decorating scenes with trees or tables.

### PR DESCRIPTION
**Pull request by WillBennett2**
- World borders (shop + farm)
- camera fixed pos in shop
- end phase button
- Phase changes
- health potions
- player on top of items
- changes to game manager
- new UI for shop that doesn't include phase changes
- Level triggers (moving room to room)
- Separate Entity files for each level 

image
image
image

Shop is broken
- cant buy things sometimes

Night can happen when no enemies have spawned -> goes insta back to day. This is bad bad.

### CODE REVIEW

File: Roche-Engine/Objects/ShopItem.cpp [hunk 1]
```diff
@@ -21,17 +21,27 @@ ShopItem::~ShopItem()
...
 	{
-		std::pair<std::string, int>* item = new std::pair<std::string, int>();
-		item->first = m_itemName;
-		item->second = 1;
-		EventSystem::Instance()->AddEvent(EVENTID::UpdateSeed, item);
-	}
+		m_bCollisionCheck = true;
 
-	m_bInputCheck = false;
+		if (m_bInputCheck)
```
Comment (charliemorris56) [inline Roche-Engine/Objects/ShopItem.cpp:28]: Bugging out?

**Review by charliemorris56 — APPROVED** [review]

File: Roche-Engine/Objects/ShopItem.cpp (hunk 1 above)
Comment (charliemorris56) [reply Roche-Engine/Objects/ShopItem.cpp:28]: will fix it later

File: Roche-Engine/LevelTrigger.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#ifndef LEVELTRIGGER_H
+#define LEVELTRIGGER_H
+
+#include <functional>
+#include "Collider.h"
+#include "EventSystem.h"
+#include "GameManager.h"
```
Comment (kyle-robinson) [inline Roche-Engine/LevelTrigger.h:9]: _NOTE: Remember to remove these._

File: Roche-Engine/Objects/ShopItem.cpp [hunk 3]
```diff
@@ -21,17 +21,27 @@ ShopItem::~ShopItem()
...
+			if (m_itemName == "HealthPotion")
+			{
+				EventSystem::Instance()->AddEvent(EVENTID::BuyPotion);
+			}
+			else
+			{
+				std::pair<std::string, int>* item = new std::pair<std::string, int>();
+				item->first = m_itemName;
+				item->second = 1;
+				EventSystem::Instance()->AddEvent(EVENTID::UpdateSeed, item);
```
Comment (kyle-robinson) [inline Roche-Engine/Objects/ShopItem.cpp:39]: _NOTE: Don't need to pass the seed value._

File: Roche-Engine/LevelTrigger.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+public:
+	LevelTrigger(const std::shared_ptr<Collider> collider);
+	~LevelTrigger();
+
+	void PlayerInTrigger(Collider& collider);
+	void PlayerOutTrigger(Collider& collider);
+private:
+	Phase m_currentGamePhase;
+	bool m_bEventFired = false;
+	virtual void HandleEvent(Event* event) override;
```
Comment (kyle-robinson) [inline Roche-Engine/LevelTrigger.h:22]: _NOTE: Remove virtual._

**Review by kyle-robinson — APPROVED** [review]
Really great work guys! Well done 😎 👍🏻

**Pull request merged**